### ISSUE TITLE
[`ReferenceSpacing`] Add support for `spacesCountBeforeReference`

### ISSUE THREAD

**Issue by carusogabriel** [slevomat/coding-standard#928]
Current, for the `SlevomatCodingStandard.PHP.ReferenceSpacing` sniff, there's only the `spacesCountAfterReference`. This option isn't enough to fix codes like:

```php
    public function referenceAssignment(): void
    {
        $foo = 1;

        $bar =& $foo;
    }
```

because they get converted to:

```php
    public function referenceAssignment(): void
    {
        $foo = 1;

        $bar =&$foo;
    }
```

when the desired output is actually:

```php
    public function referenceAssignment(): void
    {
        $foo = 1;

        $bar = &$foo;
    }
```

**Comment by kukulich** [comment]
I'm not sure if it makes sense.

1. The space should be solved in other sniff like `OperatorSpacingSniff`, `FunctionDeclarationSniff` etc.

2. You would probably need different settings for `= &$foo`, `function (&$foo)`, `use (&$foo)` etc.

That's why I implemented only `spacesCountAfterReference`.

**Comment by carusogabriel** [comment]
Yeah, indeed, by the second point this wouldn't make much sense. Thanks for the explanation 👍
